plugins/services/src/js/stores/MesosLogStore.js

```diff
@@ -97,50 +104,57 @@ class MesosLogStore extends GetSetBaseStore {
       startOffset = 0;
     }
 
-    if (startOffset === 0 && logBuffer.getStart() === 0) {
-      // Already at the top.
-      return;
-    }
-
     MesosLogActions.fetchPreviousLog(
-      slaveID, path, startOffset, MAX_FILE_SIZE
+      slaveID,
+      path,
+      startOffset,
+      MAX_FILE_SIZE
     );
   }
 
   startTailing(slaveID, path) {
-    const logBuffer = new LogBuffer();
-    this.set({[path]: logBuffer});
-    // Request offset to initialize logBuffer
-    MesosLogActions.requestOffset(slaveID, path);
+    let logBuffer = this.getLogBuffer(path);
+    if (!logBuffer) {
+      logBuffer = new LogBuffer();
+      this.logs[path] = {logBuffer, isTailing: true};
+      // Request offset to initialize logBuffer
+      MesosLogActions.requestOffset(slaveID, path);
+    } else {
+      // Continue tailing, whereever we left off
+      MesosLogActions.fetchLog(
+        slaveID,
+        path,
+        logBuffer.getEnd(),
+        MAX_FILE_SIZE
+      );
+    }
   }
 
-  stopTailing(path) {
+  stopTailing(path, shouldClearData = false) {
```

Comment:
Please rename `shouldClearData` to `clearData`  to follow the general naming schema #nit